pogobuf/pogobuf.client.js

```diff
@@ -953,23 +979,40 @@ function Client() {
         // If the requests include a map objects request, make sure the minimum delay
         // since the last call has passed
         if (requests.some(r => r.type === RequestType.GET_MAP_OBJECTS)) {
+
+            // Check distance traveled to see if it is past refresh distance
+            var refreshRequired = !self.withinInteractionRange(
+                self.lastMapObjectsLatitude,
+                self.lastMapObjectsLongitude,
+                self.mapObjectsMinDistance);
             var now = new Date().getTime(),
-                delayNeeded = self.lastMapObjectsCall + self.mapObjectsMinDelay - now;
+                delayNeeded = self.lastMapObjectsCall + self.mapObjectsMaxDelay - now;
```

Comment:
@joelfi: I would agree that the library shouldn't throttle too much, but maybe throttling should be disabled by default then?

I'm lost as to how this would break scanners though.  I'm a scanner.  I move my location when I scan.  The center cell of my new scan location is likely much further than 10 meters than the center cell of my previous scan location since I want to cover as much ground as possible with as little overlap as possible.

The old implementation of this delay logic would require me to disable throttling or wait 10 seconds before I can scan again.  The new implementation of this logic would have the same impact as if throttling was disabled, because in all likelihood the scanner position has moved more than 10 meters.

This is also similar to the logic @AeonLucid uses in his [POGOLib](https://github.com/AeonLucid/POGOLib#heartbeats)